src/main/plugin/iso19139.ca.HNAP/layout/utility-tpl-multilingual.xsl

```diff
@@ -102,7 +114,13 @@
                   <xsl:otherwise><xsl:value-of select="gmd:languageCode/gmd:LanguageCode/@codeListValue"/></xsl:otherwise>
                </xsl:choose>
             </xsl:variable>
-            <lang><xsl:value-of select="concat('&quot;', $lang_ISO639_2B, '&quot;:&quot;#', @id, '&quot;')"/></lang>
+            <xsl:variable name="id_ISO639_2B">
+              <xsl:choose>
+                <xsl:when test="@id = 'fra'">fre</xsl:when>
+                <xsl:otherwise><xsl:value-of select="@id"/></xsl:otherwise>
+              </xsl:choose>
+            </xsl:variable>
+            <lang><xsl:value-of select="concat('&quot;', $lang_ISO639_2B, '&quot;:&quot;#', $id_ISO639_2B, '&quot;')"/></lang>
```

Comment:
I don't believe I agree with this change.
The `@id`  is just a reference to the PT_Locale.  It could be called `XYZ` for French local and referenced as `#XYZ` - if you need the real language code for  `#XYZ` then you have to get it from the codeValueList of the pt_locale.  
The `@id` should just be treated as a reference/variable name.

I have seen some logic in the code which treats the `@id` as the real language and it causes bugs. So if you see the  value from the `@id` treated as the real language somewhere, it is most likely a bug.

Reply:
> The @id is just a reference to the PT_Locale. It could be called XYZ

True but note that if at some point you want to use multilingual subtemplates (which does not have a section to declare language ids) it will not work as it is now - that's why default is to build ids from the 2 first letters of the language code. We might be able to find more elegant solution for this BTW ... It mainly comes from geocat.ch in which user can only use subtemplates for contacts, extents and formats.

Reply:
I have the code to use code list instead of id here. I also added some more code fix to layout.xsl which is responsible for this element in English mode
<gmd:PT_FreeText>
       <gmd:textGroup>
           <gmd:LocalisedCharacterString locale="#fra">200 rue Kent</gmd:LocalisedCharacterString>
        </gmd:textGroup>
 </gmd:PT_FreeText>